clang/test/SemaOpenCL/amd-media-ops.cl

```diff
@@ -0,0 +1,98 @@
+// RUN: %clang_cc1 %s -triple amdgcn-unknown-unknown -verify -pedantic -Wconversion -Werror -fsyntax-only -cl-std=CL -finclude-default-header %s
+// RUN: %clang_cc1 %s -triple amdgcn-unknown-unknown -verify -pedantic -Wconversion -Werror -fsyntax-only -cl-std=CL -finclude-default-header -fdeclare-opencl-builtins %s
+// expected-no-diagnostics
+
```

Comment:
Don't you need to use the pragma to enable the extension? This is accepting these without it?

Reply:
If this extension only adds new builtin functions without altering parsing, you [shouldn't need any pragmas](https://clang.llvm.org/docs/OpenCLSupport.html#opencl-extensions-and-features).  I suppose the extension is enabled for the chosen triple already.

Reply:
I think the feature map is doing 2 things. Assigning `true` to all the supported features may be (one) of the issues here.
If an element is in the map it is supported, and if it is set to true it is enabled.

In our case there is a macro definition for cl_amd_media_ops set to 1. Later, in SemaLookup.cpp InsertOCLBuiltinDeclarationsFromTable checks only for macro definitions (and strangely not for extensions defined via macros through the OpenCLOptions class).

Reply:
I've added the test case for -cl_amd_media_ops and -cl_amd_media_ops2.